src/components/UploadLabelButton.js

```diff
@@ -0,0 +1,209 @@
+import { isFileEmpty } from '../libs/utils';
+import { DAR } from '../libs/ajax';
+import { useEffect, useState } from 'react';
+import { Notifications } from '../libs/utils';
+
+///Constant styles, not dependent on component variables
+const uploadFileLabelColors = {
+  standardBackgroundColor: 'rgb(96, 59, 155)',
+  hoverBackgroundColor: '#2FA4E7'
+};
+
+const fileClearColor = {
+  standardColors: {
+    backgroundColor: '#fdc3c3ab',
+    color: '#fb3737'
+  },
+  hoverColors: {
+    backgroundColor: 'red',
+    color: 'white'
+  },
+  transition: 'background 0.3s ease'
+};
+
+const fileNameColor = {
+  standardColors: {
+    backgroundColor: 'rgb(243 248 253)',
+    color: 'black'
+  },
+  hoverColors: {
+    backgroundColor: '#2FA4E7',
+    color: 'white'
+  },
+  transition: 'background 0.3s ease'
+};
+
+const filenameStyle = {
+  flex: 1,
+  justifyContent: 'left',
+  whiteSpace: 'nowrap',
+  minWidth: '30rem',
+  overflow: 'hidden',
+  marginLeft: '4rem',
+  fontFamily: 'Montserrant',
+  fontSize: '1.8rem',
+  padding: '0.5rem',
+  border: '1px solid #ea5e5',
+  cursor: 'pointer',
+  backgroundColor: fileNameColor.standardColors.backgroundColor,
+  color: fileNameColor.standardColors.color,
+  display: 'inline-flex',
+  alignItems: 'center',
+  borderTopLeftRadius: '2rem',
+  borderBottomLeftRadius: '2rem',
+  maxWidth: '30rem',
+  boxShadow: "-4px 6px 9px 0px #e8e5e5",
+  transition: fileNameColor.transition,
+};
+const uploadFileInput = {
+  display: 'none'
+};
+
+export default function UploadLabelButton(props) {
+  const {
+    id,
+    formAttribute,
+    newDULFile = {}, //refers to recently uploaded file (via browser upload, for reference if user cancels a future upload attempt)
+    changeDULDocument,
+    currentFileName = '', //refers to filename in db record
+    currentFileLocation = '', //use to allow user to download current file? (needs to be implemented),
+    darCode,
+    referenceId
+  } = props;
+
+  const [targetDownload, setTargetDownload] = useState(null);
+
+  useEffect(() => {
+    const downloadFile = async() => {
+      const fileAttributeLinks = {
+        irbDocument: 'irbDocument',
+        collaborationLetter: 'collaborationDocument'
+      };
+
+      if(!isFileEmpty(newDULFile)) {
+        setTargetDownload(URL.createObjectURL(newDULFile));
+      } else if(!isNil(currentFileLocation) && !isEmpty(currentFileLocation)) {
+        try{
+          let targetResponse = await DAR.getIRBDocument(referenceId, fileAttributeLinks[formAttribute]);
+          const targetFile = new Blob([targetResponse], {type: 'application/octet-stream'});
+          if(!isNil(targetFile)) {
+            setTargetDownload(URL.createObjectURL(targetFile));
+          }
+        }catch(error) {
+          Notifications.showError({text: 'Error: Failed to retreive saved file'});
+        }
+      }
+    };
+    downloadFile();
+  }, [currentFileLocation, currentFileName, formAttribute, newDULFile, referenceId]);
```

Comment:
The arguments are essentially telling the hook what variables to watch for updates. This way, instead of the hook firing off on every re-render (which can happen on any prop variable or state variable update), it'll only fire off on the values that it cares about. 

You can have an empty array as an argument if you want the hook to run only on mount, but if the process and/or result of the hook is dependent on the arguments then there's no reason to leave it empty at all.